docs/crr-to-public-cloud.md

```diff
@@ -0,0 +1,123 @@
+# CRR from CloudServer to public cloud - Workflow
+
+## Installation
+
+An endpoint ('awsbackend', for example) will be defined in the [location
+configuration](https://github.com/scality/S3/blob/master/locationConfig.json) of
+[CloudServer](https://github.com/scality/S3). Each location constraint in the
+configuration file has properties needed for performing operations for the
+location. For example, if replicating to AWS S3, the properties will include the
+bucket name, region, and credentials for AWS S3. This enables the multiple
+backend feature in CloudServer that Backbeat uses for CRR to a public cloud.
+
+Replication endpoints are defined in the config for both Backbeat and
+CloudServer in Federation's [group_vars/all](https://github.com/scality/
+Federation/blob/master/env/client-template/group_vars/all) file:
+
+```
+env_replication_endpoints:
+  site: awsbackend
+  type: aws_s3
+  site: azurebackend
+  type: azure
+```
+
+During installation, these values are set in the destination `bootstrapList` in
+Backbeat's [configuration
+file](https://github.com/scality/backbeat/blob/master/conf/config.json) and the
+`replicationEndpoints` value in CloudServer's [configuration
+file](https://github.com/scality/S3/blob/master/config.json).
+
+### Operations
+
+Below is a sequence diagram starting from the user's PUT object request on a
+bucket with replication enabled.
+
+![design](/res/queue-processor-sequence-diagram.png)
+
+#### CloudServer: PUT Bucket Replication Request
+
+* CloudServer receives the request and checks the storage class set in the
+  bucket replication configuration (see the
+  `StorageClass` property [here](https://docs.aws.amazon.com/
+  AWSJavaScriptSDK/latest/AWS/S3.html#putBucketReplication-property)).
```

Comment:
We should specify that storage class is different for us and AWS: we don't support specifying the storage class in the AWS sense for replication workflow (Standard infrquent access, for example, which ppl tend to choose as it is cheaper)